x-pack/plugins/cross_cluster_replication/public/app/sections/auto_follow_pattern_add/auto_follow_pattern_add.js

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { PureComponent, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { injectI18n, FormattedMessage } from '@kbn/i18n/react';
+
+import {
+  EuiBreadcrumbs,
+  EuiPage,
+  EuiPageBody,
+  EuiPageContent,
+  EuiPageContentHeader,
+  EuiSpacer,
+  EuiTitle,
+  EuiButton,
+  EuiCallOut,
+} from '@elastic/eui';
+
+import routing from '../../services/routing';
+import { BASE_PATH_REMOTE_CLUSTERS } from '../../../../common/constants';
+import { AutoFollowPatternForm, RemoteClustersProvider, SectionLoading, SectionError } from '../../components';
+
+class AutoFollowPatternAddUI extends PureComponent {
+  static propTypes = {
+    createAutoFollowPattern: PropTypes.func.isRequired,
+    clearApiError: PropTypes.func.isRequired,
+    apiError: PropTypes.object,
+    apiStatus: PropTypes.string.isRequired,
+  }
+
+  componentWillUnmount() {
+    this.props.clearApiError();
+  }
+
+  renderEmptyClusters() {
+    const { intl } = this.props;
+    const title = intl.formatMessage({
+      id: 'xpack.crossClusterReplication.autoFollowPatternCreateForm.emptyRemoteClustersCallOutTitle',
+      defaultMessage: 'No remote cluster found'
+    });
+
+    return (
+      <Fragment>
+        <EuiCallOut
+          title={title}
+          color="warning"
+          iconType="cross"
+        >
+          <FormattedMessage
+            id="xpack.crossClusterReplication.autoFollowPatternCreateForm.emptyRemoteClustersCallOutDescription"
+            defaultMessage="You haven't added yet any remote cluster. In order to create an auto-follow pattern you need to add a remote cluster." //eslint-disable-line max-len
+          />
+        </EuiCallOut>
+        <EuiSpacer />
+        <EuiButton
+          {...routing.getRouterLinkProps('/add', BASE_PATH_REMOTE_CLUSTERS)}
+          fill
+          iconType="plusInCircle"
+        >
+          <FormattedMessage
+            id="xpack.crossClusterReplication.autoFollowPatternCreateForm.addRemoteClusterButtonLabel"
+            defaultMessage="Add a remote cluster"
+          />
+        </EuiButton>
+      </Fragment>
+    );
+  }
+
+  renderNoConnectedCluster() {
+    const { intl } = this.props;
+    const title = intl.formatMessage({
+      id: 'xpack.crossClusterReplication.autoFollowPatternCreateForm.noRemoteClustersConnectedCallOutTitle',
+      defaultMessage: 'Remote cluster connection error'
+    });
+
+    return (
```

Comment:
How about moving the button inside of the callout? This way the problem and solution are associated more closely with one another. Right now the EUI callout styles mess up the color of the button but I can help fix this in EUI if you think this is an improvement. I also made some changes to the copy to make it a bit clearer.

![image](https://user-images.githubusercontent.com/1238659/49321675-b952ae80-f4bd-11e8-8aee-b3441f0b3d62.png)

```js
    return (
      <EuiCallOut
        title={title}
        color="warning"
        iconType="cross"
      >
        <p>
          <FormattedMessage
            id="xpack.crossClusterReplication.autoFollowPatternCreateForm.emptyRemoteClustersCallOutDescription"
            defaultMessage="Auto-follow patterns capture indices on remote clusters. Add a remote cluster before you create an auto-follow pattern." //eslint-disable-line max-len
          />
        </p>

        <EuiButton
          {...routing.getRouterLinkProps('/add', BASE_PATH_REMOTE_CLUSTERS)}
          iconType="plusInCircle"
          color="warning"
        >
          <FormattedMessage
            id="xpack.crossClusterReplication.autoFollowPatternCreateForm.addRemoteClusterButtonLabel"
            defaultMessage="Add remote cluster"
          />
        </EuiButton>
      </EuiCallOut>
    );
```